worker/lease/manager.go

```diff
@@ -515,38 +513,38 @@ func (manager *Manager) checkBlocks(blocks blocks) {
 			blocks.unblock(leaseName)
 		}
 	}
-	manager.computeNextTimeout(now, leases)
+	manager.computeNextTimeout(leases)
 }
 
 // computeNextTimeout iterates the leases and finds out what the next time we
 // want to wake up, expire any leases and then handle any unblocks that happen.
-// It is based on the MaxSleep time, and on any expire that is going to expire
-// after now but before MaxSleep.
-// it's worth checking for stalled collaborators.
-func (manager *Manager) computeNextTimeout(lastTick time.Time, leases map[lease.Key]lease.Info) {
+// It is the earliest lease expiration due in the future, but before MaxSleep.
+func (manager *Manager) computeNextTimeout(leases map[lease.Key]lease.Info) {
 	now := manager.config.Clock.Now()
 	nextTick := now.Add(manager.config.MaxSleep)
 	for _, info := range leases {
-		if !info.Expiry.After(lastTick) {
-			// The previous expire will expire this lease eventually,
-			// or the manager will die with an error.
-			// Either way, we don't need to worry about expirations for
-			// a previous tick here.
-			continue
-		}
 		if info.Expiry.After(nextTick) {
 			continue
 		}
 		nextTick = info.Expiry
 	}
-	manager.config.Logger.Tracef("[%s] next expire in %v %v",
-		manager.logContext, nextTick.Sub(now).Round(time.Millisecond), nextTick)
-	manager.setNextTimeout(nextTick)
-}
 
-func (manager *Manager) setupInitialTimer() {
-	// lastTick has never happened, so pass in the epoch time
-	manager.computeNextTimeout(time.Time{}, manager.config.Store.Leases())
+	// If we had leases set to expire in the past, then we assume that our FSM
+	// is behind the leader and will soon indicate their expiration.
+	// Check the blocks again soon.
+	if !nextTick.After(now) {
+		nextTick = now
+	}
+
+	// The lease clock ticks *at least* a second from now. Expirations only
+	// occur when the global clock updater ticks the clock, so this avoids
+	// too frequently checking with the potential of having no work to do.
+	// The blanket addition of a second is no big deal.
+	nextTick.Add(time.Second)
+
+	nextDuration := nextTick.Sub(now).Round(time.Millisecond)
+	manager.config.Logger.Tracef("[%s] next expire in %v %v", manager.logContext, nextDuration, nextTick)
```

Comment:
The only thing I can think of for how we are seeing 
```
2021-08-18 19:59:52 TRACE juju.worker.lease.raft manager.go:546 [0a61b5] next expire in 966ms 2021-08-18 19:59:53.397286643 +0000 UTC m=+15976.527887156
```

the only thing I could think of is if there is a discrepancy between monotonic clock and global clock of a given time.